common/utxo.h

```diff
@@ -26,6 +26,9 @@ struct utxo {
 	/* Optional unilateral close information, NULL if this is just
 	 * a HD key */
 	struct unilateral_close_info *close_info;
+
+	/* Blockchain height references */
+	const int *confirmation_height, *spend_height;
```

Comment:
I'd like a bit more description for these fields; and we use u32 for block heights (the other option is to make these non-pointers and use -1, but I like the crash we get if they're NULL and we make a mistake):

/* NULL if we haven't seen it in a block, otherwise the block it's in */
const u32 *confirmation_height;

/* NULL if not spent yet, otherwise, the block the spending transaction is in */
const u32 *spend_height;

Reply:
Also, 'confirmation_height' here and in db could also be called 'block_height'?  The word confirmation is usually used with depth, not height, so it seems a bit awkward here?  In other places you use blockheight and spendheight, it seems.

Reply:
`confirmed_at_height` and `spent_at_height`?

Reply:
Now using `blockheight` and `spendheight`, consistent with the utxoset nomenclature. The `block->height` is of type `int`, I'll change that in a cleanup PR.